src/test-provider/test-item-data.ts

```diff
@@ -175,15 +176,24 @@ export class WorkspaceRoot extends TestItemDataBase {
   }
   createTestItem(): vscode.TestItem {
     const workspaceFolder = this.context.ext.workspace;
+    const settings = this.context.ext.settings;
+    let uri = isVirtualWorkspaceFolder(workspaceFolder)
+      ? workspaceFolder.effectiveUri
+      : workspaceFolder.uri;
+    if (settings.trimSymlinks) {
+      // In case the workspace root (or one of its ancestors) is a symlink, the relative path is going to resolve
+      // up-levels (../) until the first common ancestor with the link, resulting in many nodes we can hide from the user.
+      // In order to hide them, we get the workspaceRoot's "realpath" and use it instead.
+      uri = vscode.Uri.file(realpathSync(uri!.fsPath));
```

Comment:
To be safe, maybe let's wrap it inside a try/catch block and fall back to the original uri...